src/js/components/visualization/Visualization.jsx

```diff
@@ -93,6 +93,68 @@ class Visualization extends Component {
       typeGroup
     }
   }
+  
+  renderError (error) {
```

Comment:
I see the renderError() method becoming a utility that will need the error passed to it.